src/test/kotlin/no/nav/syfo/infrastructure/cronjob/RepublishForhandsvarselWithAdditionalInfoCronjobSpek.kt

```diff
@@ -0,0 +1,159 @@
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.concurrent.Future
+
+class RepublishForhandsvarselWithAdditionalInfoCronjobSpek : Spek({
+    describe(RepublishForhandsvarselWithAdditionalInfoCronjob::class.java.simpleName) {
+        val externalMockEnvironment = ExternalMockEnvironment.instance
+        val database = externalMockEnvironment.database
+
+        val vurderingRepository = VurderingRepository(database = database)
+        val vurderingPdfService = VurderingPdfService(
+            externalMockEnvironment.pdfgenClient,
+            externalMockEnvironment.pdlClient,
+        )
+        val journalforingService = JournalforingService(
+            dokarkivClient = externalMockEnvironment.dokarkivClient,
+            pdlClient = externalMockEnvironment.pdlClient,
+            isJournalforingRetryEnabled = externalMockEnvironment.environment.isJournalforingRetryEnabled,
+        )
+
+        val mockVurderingProducer = mockk<KafkaProducer<String, VurderingRecord>>(relaxed = true)
+        val vurderingProducer = VurderingProducer(
+            producer = mockVurderingProducer,
+        )
+
+        val vurderingService = VurderingService(
+            vurderingRepository = vurderingRepository,
+            vurderingPdfService = vurderingPdfService,
+            journalforingService = journalforingService,
+            vurderingProducer = vurderingProducer,
+        )
+
+        beforeEachTest {
+            clearAllMocks()
+            coEvery { mockVurderingProducer.send(any()) } returns mockk<Future<RecordMetadata>>(relaxed = true)
+        }
+
+        afterEachTest {
+            database.dropData()
+        }
+
+        describe("RepublishForhandsvarselWithAdditionalInfoCronjob") {
+            val begrunnelse1 = "En begrunnelse1"
+            val svarfrist1 = LocalDate.of(2025, 3, 1)
+            val vurderingForhandsvarsel1 = generateForhandsvarselVurdering(
+                begrunnelse = begrunnelse1,
+                document = generateForhandsvarselRevarslingDocumentComponent(
+                    beskrivelse = begrunnelse1,
+                    svarfrist = svarfrist1,
+                ),
+                svarfrist = svarfrist1
+
+            )
+            val begrunnelse2 = "En begrunnelse2"
+            val svarfrist2 = LocalDate.of(2025, 3, 7)
+            val vurderingForhandsvarsel2 = generateForhandsvarselVurdering(
+                personident = ARBEIDSTAKER_2_PERSONIDENT,
+                begrunnelse = begrunnelse2,
+                document = generateForhandsvarselRevarslingDocumentComponent(
+                    beskrivelse = begrunnelse2,
+                    svarfrist = svarfrist2,
+                ),
+                svarfrist = svarfrist2
+            )
+
+            it("Henter gamle, overskriver teksten og svarfrist") {
```

Comment:
Kan også legge på en test for hvis man sender inn en uuid som ikke finnes i databasen og se hvordan det oppfører seg når det feiler.